ivy/functional/backends/paddle/experimental/elementwise.py

```diff
@@ -83,13 +83,18 @@ def float_power(
     return paddle.cast(paddle.pow(x1, x2), dtype=paddle.float64)
 
 
+@with_unsupported_device_and_dtypes(
+    {"2.4.2 and below": {"cpu": ("uint16", "bfloat16","float16")}}, backend_version
+)  
 def exp2(
     x: Union[paddle.Tensor, float, list, tuple],
     /,
     *,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    y = paddle.full(shape=x.shape, fill_value=2, dtype=x.dtype)
+    res = paddle.pow(y,x)
```

Comment:
No need to create a full array with 2s since it's automatically broadcasted, just pass 2 as a tensor, also the following data types are not supported: `'int8', 'int16', 'uint8', 'float16', 'complex64', 'complex128', 'bool'` so I suggest using ivy.pow in array mode instead:
```
with ivy.ArrayMode(False):
    return ivy.pow(2,x)
```